source/common/network/connection_impl.cc

```diff
@@ -671,6 +671,7 @@ void ConnectionImpl::onWriteReady() {
     if (error == 0) {
       ENVOY_CONN_LOG(debug, "connected", *this);
       connecting_ = false;
+      onConnected();
```

Comment:
Only client connection will hit this branch. 

Although the server connection's onConnected() is no-op for now, I suggest adding the onConnected() [here ](https://github.com/envoyproxy/envoy/blob/main/source/common/network/connection_impl.cc#L85-L87)

Reply:
This interval tracks the time it took from the `connect` call to connection being connected. This can only be applicable to upstream/client connections.